enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,635 @@
+---
+title: automated-resource-request-scaling
+authors:
+  - "@dhellmann"
+  - "@csrwng"
+  - "@wking"
+  - "@browsell"
+reviewers:
+  - TBD
+approvers:
+  - "@smarterclayton"
+  - "@derekwaynecarr"
+  - "@markmc"
+creation-date: 2021-02-04
+last-updated: 2021-02-04
+status: provisional
+see-also:
+  - "/enhancements/single-node-production-deployment-approach.md"
+---
+
+# Automated Resource Request Scaling for Control Plane
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes an approach to allow us to scale the
+resource requests for the control plane services to reduce consumption
+for constrained environments. This will be especially useful for
+single-node production deployments, where the user wants to reserve
+most of the CPU resources for their own workloads and needs to
+configure OpenShift to run on a fixed number of CPUs within the host.
+
+One example of this use case is seen in telecommunication service
+providers implementation of a Radio Access Network (RAN). This use case
+is discussed in more detail below.
+
+## Motivation
+
+The resource requests for cluster operators and their operands are
+based on performance analysis on multiple cloud VMs using the
+end-to-end test suite to gather data. While the resulting numbers work
+well for similar cloud environments and even multi-node bare metal
+deployments, they result in over-provisioning resources for
+single-node deployments.
+
+### Goals
+
+* This enhancement describes an approach for configuring OpenShift
+  clusters to run with lower CPU resource requests than its default
```

Comment:
We should probably broaden this or qualify this as "compressible resources" - by definition this wouldn't help with incompressible resources.

Reply:
I clarified that the goal of this enhancement is to only address CPU, and that we would look at approaches for other compressible resources in other enhancements. I've given the doc a new title, but was reluctant to rename the file and lose all of the review threads. I will do that before we merge it, or in another PR.